### PR TITLE
fix(forge): No bail on setup

### LINKED ISSUE #433: Stack Trace on setUp() fail
If `setUp()` reverts, forge will panic instead of printing out the revert stack trace. This is sub-optimal for development as reverts are quite common as your building out tests. Would prefer this to output the error as if the revert was in the test itself.

### PR DESCRIPTION
**Pull request by brockelmore**
Instead of bailing when a `setUp` fails, we just return a failed test. This allows people to debug the setup via `-vvv` and generally just know why the setup failed.

Closes #433

### CODE REVIEW

File: forge/src/multi_runner.rs [hunk 1]
```diff
@@ -220,8 +220,8 @@ mod tests {
         let mut runner = runner(evm);
         let results = runner.test(&Filter::new(".*", ".*")).unwrap();
 
-        // 6 contracts being built
-        assert_eq!(results.keys().len(), 7);
+        // 8 contracts being built
+        assert_eq!(results.keys().len(), 8);
```
Comment (onbjerg) [inline forge/src/multi_runner.rs:224]: think this is because of the regression mentioned in 4bdff194667accbb06a83e307c7e3cb05b0b4282

Comment (brockelmore) [reply forge/src/multi_runner.rs:224]: my contract isnt abstract so it should still increment by 1, no?

Comment (onbjerg) [reply forge/src/multi_runner.rs:224]: mb :facepalm:

Comment (gakonst) [reply forge/src/multi_runner.rs:224]: The regression should be unrelated, this seems correct, there's a new contract added `BadSetup` so makes sense that there's 1 more key.

**Review by gakonst — CHANGES_REQUESTED** [review]
Change LGTM - should we also add a test?

**Comment by brockelmore** [comment]
whats best way to fix this? the setup should fail, which will always cause tests to fail, regardless if they are `testFail`. Should we filter out the contract? from the results iter?

**Comment by gakonst** [comment]
Yeah let's just filter it out for now. Think we'll want to segregate the compilation of each set of test vectors eventually, but OK for now.

**Comment by brockelmore** [comment]
test failure was a flake, this should be gtg

**Review by gakonst — APPROVED** [review]

**Pull request merged**